recipe-server/client/control_new/state/router/selectors.js

```diff
@@ -7,3 +7,24 @@
 export function getRouterParam(state, paramName) {
   return state.router.params[paramName];
 }
+
+export function getQueryParameter(state, key, defaultsTo) {
+  return state.router.query[key] || defaultsTo;
+}
+
+export function getCurrentURL(state, queryParams) {
+  const query = {
+    ...state.router.query,
+    ...queryParams,
+  };
+
+  let queryString = '';
+  Object.keys(query).forEach(key => {
+    if (query[key]) {
+      queryString += queryString ? '&' : '?';
+      queryString += `${key}=${query[key]}`;
+    }
+  });
+
+  return `${state.router.pathname}${queryString}`;
```

Comment:
redux-little-router has a `router.search` property which I'm pretty sure makes a majority of this function redundant.